### PR TITLE
hamming: update tests to version 2.0.1

### LINKED ISSUE #1015: hamming: update tests to version 2.0.1
As identified in #784, some of the exercise test-suites are out-of-date, `hamming` is one of them. It would therefore be really helpful for people to go through and update the test suite to match the canonical data.

Tests should typically be ordered according to their order in the canonical data, since this makes it easier to check and update them in future. If you think that there's a better order, please let us know your reasoning in your PR.

If there are some additional tests that currently exist in this track but which aren't in the canonical data, these can be kept if you think that they are useful. They should be placed at the bottom of the test cases, and it would also be super-helpful if you could add a comment before them saying `# Additional tests for this track` to make it clear that they aren't part of the canonical data.

The version string should be placed just before the first test class and followed by a blank line. For this exercise, the version string is:
```
# Tests adapted from `problem-specifications//canonical-data.json` @ v2.0.1
```

**Test suite:** [exercism/python/exercises/hamming/hamming_test.py](https://github.com/exercism/python/blob/master/exercises/hamming/hamming_test.py)
**Canonical data:** [exercism/problem-specifications/exercises/hamming/canonical-data.json](https://github.com/exercism/problem-specifications/blob/master/exercises/hamming/canonical-data.json)

### PR DESCRIPTION
**Pull request by kishan3**
Resolves #1015

### CODE REVIEW

**Comment by N-Parsons** [comment]
If you reference the corresponding issue with `Resolved #issue` (eg. `Resolves #1015` in this case) to your pull request description, it helps us to keep track of which issues are going to be resolved by each pull request, and avoids having multiple pull requests for an issue.

**Comment by kishan3** [comment]
Resolves #1015

**Comment by N-Parsons** [comment]
@kishan3, I believe that the [magic words](https://help.github.com/articles/closing-issues-using-keywords/) need to be in the description for them to work. You can edit your description by clicking the pencil icon on the top-right of your first comment, where it currently says _"No description provided."_.

**Comment by kishan3** [comment]
Sure sir thank you for guidance.

**Comment by kishan3** [comment]
Hello @N-Parsons why the check is failing can you please help me understand?

File: exercises/hamming/hamming_test.py [hunk 1]
```diff
@@ -3,10 +3,13 @@
 import hamming
 
 
-# test cases adapted from `x-common//canonical-data.json` @ version: 1.0.0
+# Tests adapted from `problem-specifications//canonical-data.json` @ v2.0.1
 
 class HammingTest(unittest.TestCase):
-
+
```
Comment (N-Parsons) [inline exercises/hamming/hamming_test.py]: You have extra whitespace in this line.

File: exercises/hamming/hamming_test.py [hunk 2]
```diff
@@ -3,10 +3,13 @@
...
 
-# test cases adapted from `x-common//canonical-data.json` @ version: 1.0.0
+# Tests adapted from `problem-specifications//canonical-data.json` @ v2.0.1
 
 class HammingTest(unittest.TestCase):
-
+    
+    def test_identical_strands(self):
+        self.assertEqual(hamming.distance("", ""), 0)
+
```
Comment (N-Parsons) [inline exercises/hamming/hamming_test.py]: You have extra whitespace in this line.

File: exercises/hamming/hamming_test.py [hunk 3]
```diff
@@ -3,10 +3,13 @@
 import hamming
 
 
-# test cases adapted from `x-common//canonical-data.json` @ version: 1.0.0
+# Tests adapted from `problem-specifications//canonical-data.json` @ v2.0.1
 
 class HammingTest(unittest.TestCase):
-
+    
+    def test_identical_strands(self):
```
Comment (N-Parsons) [inline exercises/hamming/hamming_test.py]: This function shares it's name with the one below, so it's just getting redefined without use. You need to change the name of one of these functions.

**Comment by kishan3** [comment]
Thank you sir @N-Parsons.  Will make the changes.

**Comment by kishan3** [comment]
@N-Parsons  Fixed Failing Builds Please check.

**Comment by kishan3** [comment]
Sure @N-Parsons. Thanks making the changes.

**Comment by kishan3** [comment]
Updated.

**Comment by N-Parsons** [comment]
Perfect! Thanks @kishan3!

**Pull request merged**